src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -85,21 +93,114 @@ public function setCallback($callback = null)
     }
 
     /**
+     * Sets the content to be sent as JSON.
+     *
+     * @param string $content
+     *
+     * @return Response
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function setContent($content)
+    {
+        $data = json_decode($content);
+        if (JSON_ERROR_NONE !== json_last_error()) {
+            throw new \InvalidArgumentException(json_last_error_msg());
+        }
+        $this->data = $data;
+
+        return $this->update();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContent()
+    {
+        if ($this->callback !== null) {
+            // Not using application/javascript for compatibility reasons with older browsers.
+            $this->headers->set('Content-Type', 'text/javascript');
+
+            return sprintf('/**/%s(%s);', $this->callback, $this->content);
+        }
+
+        return parent::getContent();
+    }
+
+    /**
+     * Add content to existing data array.
+     *
+     * @param array $data
+     *
+     * @return JsonResponse
+     *
+     * @throws \Exception
+     */
+    public function addData($data = array())
+    {
+        $this->data = array_merge_recursive($this->data, (array) $data);
+
+        return $this->update();
+    }
+
+    /**
      * Sets the data to be sent as JSON.
      *
      * @param mixed $data
      *
      * @return JsonResponse
-     *
-     * @throws \InvalidArgumentException
      */
     public function setData($data = array())
     {
+        $this->data = $data;
```

Comment:
you are changing the content of ``$this->data`` here, which is a BC break as the property is protected